roqoqo/tests/integration/operations/pragma_operations.rs

```diff
@@ -3109,3 +3109,176 @@ fn pragma_change_device_substitute_trait() {
     let result = pragma_test.remap_qubits(&qubit_mapping_test).is_err();
     assert!(result);
 }
+
+/// Test PragmaConditional inputs and involved qubits
+#[test]
+fn pragma_controlled_circuit_inputs_qubits() {
+    let mut circuit = Circuit::new();
+    circuit.add_operation(PauliX::new(0));
+    let pragma = PragmaControlledCircuit::new(1, circuit.clone());
+
+    // Test inputs are correct
+    assert_eq!(pragma.controlling_qubit(), &1_usize);
+    assert_eq!(pragma.circuit(), &circuit.clone());
+
+    // Test InvolveQubits trait
+    let mut qubits: HashSet<usize> = HashSet::new();
+    qubits.insert(0);
+    qubits.insert(1);
+    assert_eq!(pragma.involved_qubits(), InvolvedQubits::Set(qubits));
+}
+
```

Comment:
I would suggest adding a couple of lines testing the `All` and `None` variants of the control circuit's InvolvedQubits

Reply:
Added tests.